onnxruntime/core/optimizer/initializer.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/initializer.h"
+
+#include "gsl/gsl"
+
+#include "core/common/path.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/framework/tensor_external_data_info.h"
+#include "core/platform/env.h"
+
+namespace onnxruntime {
+Status Initializer::ReadExternalRawData(
```

Comment:
Looks like there is a lot of overlap between this and the code in tensorprotoutils.cc:TensorProtoToMLValue. Can it be refactored?

Reply:
the overlap i see is mainly calls to ExternalDataInfo::Create() and Env::ReadFileIntoBuffer(). what do you have in mind?

Reply:
I had a chat with Ed last evening. The whole functionality around reading the serialized tensorproto is repeated in initializer.h which should be refactored to reuse from tensorprotoutils.h. Doesn't have to be this PR as this will require a good amount of work.

Reply:
Right. This can be done in an other PR

---
In reply to: [382261519](https://github.com/microsoft/onnxruntime/pull/3045#discussion_r382261519) [](ancestors = 382261519)